ga4gh/decorators.py

```diff
@@ -0,0 +1,92 @@
+"""
+General purpose decorators
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import functools
+import signal
+import time
+
+
+class TimeoutException(Exception):
+    """
+    A process has taken too long to execute
+    """
+
+
+class Repeat(object):
+    """
+    A decorator to use for repeating a tagged function.
```

Comment:
Ah, I see, the end-to-end test-case. I've been meaning to think about this one a bit more, but perhaps we could avoid this trickiness by keeping things in the same process space? Is there some way we could wire the client and the server together without going over a socket explicitly? (Sorry, this should be on the other thread -- feel free to answer this question over there if you prefer)